### PR TITLE
Update to .NET SDK 6.0.100-preview.5

### PR DESCRIPTION
**Pull request by bricelam**

### CODE REVIEW

File: global.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "tools": {
-    "dotnet": "6.0.100-preview.4.21255.9",
+    "dotnet": "6.0.100-preview.5.21302.13",
```
Comment (bricelam) [inline global.json:3]: TODO: Use the actual, released numbers when it's out

File: eng/Version.Details.xml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <Dependencies>
   <ProductDependencies>
-    <!-- NB: Pinned for compatibility with Preview 4 -->
+    <!-- NB: Pinned for compatibility with Preview 5 -->
```
Comment (dougbu) [inline eng/Version.Details.xml:4]: Curious how this works given you're not changing these versions here and they're newer than either preview 4 or 5.

**Review by dougbu — APPROVED** [review]

File: eng/Version.Details.xml (hunk 2 above)
Comment (bricelam) [reply eng/Version.Details.xml:4]: Usually, things are highly forwards compatible with the runtime--even between prereleases. Until they're not. 😬

Comment (bricelam) [reply eng/Version.Details.xml:4]: Think of it this way: There's usually a lag of a few releases between when an API is added to the runtime and when libraries start actually taking a dependency on it.

**Pull request merged**